src/lib/classes/QContext.svelte.ts

```diff
@@ -1,26 +1,61 @@
 import { getContext, setContext } from "svelte";
 
-export class QContext<T> {
+/**
+ * This class allows to manipulate reactive contexts more easily.
+ * It avoid having to pass a Svelte store down the components but rather use runes to keep the context reactive.
+ *
+ * This class should be used when the context has to be modified from a child component. Otherwise, using svelte's context API should be enough.
+ */
+export default class QContext<T> {
   #state = $state<T>();
 
-  constructor(init: T) {
+  constructor(contextName: string, init: T) {
     this.#state = init;
+
+    setContext(contextName, this);
   }
 
+  /**
+   * Gets the value of context with id `contextName`
+   * @param contextName - Name of the context to get
+   * @returns The context's value
+   */
   static get<T>(contextName: string) {
-    return getContext<QContext<T>>(contextName);
+    return getContext<QContext<T> | undefined>(contextName);
   }
 
-  set(contextName: string) {
-    setContext(contextName, this);
-
-    return this;
+  /**
+   * Prevents the propagation of the context further down
+   * @param contextName - Name of the context to get
+   * @returns The context result
+   */
+  static reset(contextName: string) {
+    setContext(contextName, undefined);
   }
 
+  /**
+   * Updates the context's inner value with `newVal`
+   * @param newVal - New value to update the context with
+   */
   update(newVal: T) {
     this.#state = newVal;
   }
 
+  /**
+   * Updates the property of id `key` with `newVal` in the context's inner value
+   * @param newVal - New value to update the context's property with
+   */
+  updateEntry(key: keyof T, newVal: NonNullable<T>[keyof T]) {
+    if (!this.#state![key]) {
```

Comment:
What if `this.#state.someKey === false` (or `null`, `0`, etc.)? Also: Do we really want to prevent setting values for new keys?

Reply:
I should probably write `if(this.#state?.[key] === undefined)` then

Reply:
Usually, you'd want to use [`Object.hasOwn()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/hasOwn) then. You'd have to try if it works here.